pkg/v1/tkg/clusterclient/clusterclient.go

```diff
@@ -1575,6 +1579,26 @@ func (c *client) GetPacificTKCAPIVersion() (string, error) {
 	return strings.TrimSpace(match[1]), nil
 }
 
+// IsVSphere8BasedTKGSCluster returns whether the cluster is vSphere8 based TKGS cluster or not
+// Currently, we are determining this based on the presence of featuregates API
+// For vSphere 7 and older version of TKGS cluster this API will not exists but it will be available in
+// vSphere 8 based TKGS cluster
+func (c *client) IsVSphere8BasedTKGSCluster() (bool, error) {
+	// Since we're looking up API types via discovery, we don't need the dynamic client.
+	clusterQueryClient, err := capdiscovery.NewClusterQueryClient(c.dynamicClient, c.discoveryClient)
+	if err != nil {
+		return false, err
+	}
+
+	var queryObject = capdiscovery.Group("featuregates", configv1alpha1.GroupVersion.Group).WithResource("featuregates")
```

Comment:
suggestion: I would be specific about versions using `WithVersions("v1alpha1")` for some future proofing in case FeatureGates has an API version bump.

Reply:
Do we really need to care considering we just want to check if the CRD is present or not? 
The current function name seems generic, so adding `WithVersions("v1alpha1")` will require change in the CLI whenever we bump the API version.

And if the purpose for using this function is to just check presence of CRD, does it make sense to add version?

Reply:
I think you're right and it's not really necessary since vSphere 7 won't ever have these CRDs.